src/WaCore.Crud.ListSample1/Startup.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using WaCore.Crud.ListSample1.Data;
+using WaCore.Crud.ListSample1.Data.Repositories;
+using Microsoft.EntityFrameworkCore;
+using WaCore.Crud.ListSample1.Services;
+
+namespace WaCore.Crud.ListSample1
+{
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+
+            var connection = @"Server=(localdb)\mssqllocaldb;Database=WaCore.Sample;Trusted_Connection=True;";
+            services.AddDbContext<LibraryDbContext>(optionsBuilder =>
+                optionsBuilder.UseSqlServer(connection));
+
+            services.AddUnitOfWork<LibraryDbContext, IUnitOfWork, UnitOfWork>(repoConfig =>
+                repoConfig.AddRepository<IBooksListRepository, BookListRepository>());
+
+            services.AddTransient<IBookListDataService, BookListDataService>();
+            services.AddTransient<IBooksListRepository, BookListRepository>();
```

Comment:
This way you're going against the idea that we get repositories from unit of work, and not directly from DI.
You could instead configure it like this:
```
services.AddTransient<IBooksListRepository>(
    serviceProvider => serviceProvider.GetService<IUnitOfWork>().GetRepository<IBooksListRepository>());
```
Or maybe even better - we can make it so that services.AddUnitOfWork will register repositories like this.

Reply:
You're right, well spotted. I did it like you suggested, but I don't know how to implement it the way you suggest in your last sentence.

Reply:
I thought about adding another generic parameter for RepositoryConfiguration: TUnitOfWorkService and then in `public RepositoryConfiguration<TDbContext> AddRepository<TService, TImplementation>()` add line
`_serviceCollection.AddTransient<TImplementation>(serviceProvider => serviceProvider.GetRequiredService<TUnitOfWorkService>().GetRepository<TImplementation>());`
That way we you can have repositories in constructor parameters of your services and you don't have to register them in DI twice in your Startup.